File: backend/src/Web/Endpoints/ExtraCharges.cs
using CleanArchitecture.Application.ExtraCharges.Commands.CreateExtraCharge;
using CleanArchitecture.Application.ExtraCharges.Commands.DeleteExtraCharge;
using CleanArchitecture.Application.ExtraCharges.Commands.MarkAsPaid;
using Microsoft.AspNetCore.Http.HttpResults;

namespace CleanArchitecture.Web.Endpoints;

public class ExtraCharges : EndpointGroupBase
{
    public override string? GroupName => "extra-charges";

    public override void Map(RouteGroupBuilder group)
    {
        group.RequireAuthorization();

        group.MapPost("", CreateExtraCharge);
        group.MapDelete("{reservationId}/charges/{id}", DeleteExtraCharge);
        group.MapPatch("{reservationId}/charges/{id}/pay", MarkExtraChargeAsPaid);
    }

    public async Task<Created<int>> CreateExtraCharge(ISender sender, CreateExtraChargeCommand command)
    {
        var result = await sender.Send(command);
        return TypedResults.Created($"/api/extra-charges/{result}", result);
    }

    public async Task<Results<NoContent, NotFound>> DeleteExtraCharge(ISender sender, int reservationId, int id)
    {
        await sender.Send(new DeleteExtraChargeCommand(reservationId, id));
        return TypedResults.NoContent();
    }

    public async Task<Results<NoContent, NotFound>> MarkExtraChargeAsPaid(ISender sender, int reservationId, int id)
    {
        await sender.Send(new MarkExtraChargeAsPaidCommand(reservationId, id));
        return TypedResults.NoContent();
    }
}

File: front/hotel-frontend/src/api/extraCharges.ts
import http from './http';
import type { CreateExtraChargeCommand } from './types/extraCharges';

/**
 * ExtraCharges API wrapper.
 * Backend: POST /api/extra-charges, DELETE /api/extra-charges/{id}
 */

export const createExtraCharge = async (command: CreateExtraChargeCommand): Promise<number> => {
    const response = await http.post<number>('extra-charges', command);
    return response.data;
};

export const deleteExtraCharge = async (id: number): Promise<void> => {
    await http.delete(`extra-charges/${id}`);
};
